src/lm_buddy/jobs/evaluation/hf_evaluate.py

```diff
@@ -0,0 +1,115 @@
+    # (we don't want to lose all eval data if there is an issue wth s3)
+    local_path = Path(LM_BUDDY_RESULTS_PATH) / config.name / "eval_results.json"
+    local_path.parent.mkdir(exist_ok=True, parents=True)
+    with local_path.open("w") as f:
+        json.dump(evaluation_results, f)
+
+    # copy to s3 and return path
+    if storage_path is not None and storage_path.startswith("s3://"):
+        s3 = s3fs.S3FileSystem()
+        if storage_path.endswith("/"):
+            storage_path = "s3://" + str(Path(storage_path[5:]) / config.name / "eval_results.json")
+        logger.info(f"Storing into {storage_path}...")
+        s3.put_file(local_path, storage_path)
+        return storage_path
+    else:
+        return local_path
+
+
+def run_eval(config: HuggingFaceEvalJobConfig) -> Path:
+    # Init loaders
+    hf_dataset_loader = HuggingFaceDatasetLoader()
+    hf_model_loader = HuggingFaceModelLoader()
+
+    # Load dataset given its URI
+    dataset = hf_dataset_loader.load_dataset(config.dataset)
+
+    # Limit dataset length if max_samples is specified
+    if config.evaluation.max_samples is not None:
+        dataset = dataset.select(range(config.evaluation.max_samples))
+
+    # Enable / disable tqdm
+    input_samples = dataset["examples"]
+    dataset_iterable = tqdm(input_samples) if config.evaluation.enable_tqdm else input_samples
+    predictions = []
+
+    # Choose which model client to use
+    if type(config.model) == VLLMCompletionsConfig:
+        model_name = config.model.inference.base_url
+    else:
+        model_name = hf_model_loader.resolve_asset_path(config.model.path)
+
+    if model_name.startswith("http"):
+        # run the openai client
+        logger.info(f"Using OAI client. Endpoint: {model_name}")
+        model_client = OpenAIModelClient(model_name, config.model)
+    else:
+        # depending on config, use the summarizer pipeline or directly call the model
+        # for inference
+        if config.evaluation.use_pipeline:
+            logger.info(f"Using summarization pipeline. Model: {model_name}")
+            model_client = PipelineModelClient(model_name, config.model)
+        else:
+            logger.info(f"Using direct HF model invocation. Model: {model_name}")
+            model_client = HuggingFaceModelClient(model_name, config)
+
+    # run inference
+    t = time.time()
+    for sample_txt in dataset_iterable:
+        predictions.append(model_client.predict(sample_txt))
+    summarization_time = time.time() - t
+    logger.info(f"Summarization performed in {summarization_time} seconds")
+
+    # run evaluation
+    ground_truth = dataset["ground_truth"]
+    em = EvaluationMetrics(config.evaluation.metrics)
+    t = time.time()
+    evaluation_results = em.run_all(predictions, ground_truth)
+    summarization_time = time.time() - t
+    logger.info(f"Summarization performed in {summarization_time} seconds")
+
+    return save_outputs(config, evaluation_results)
```

Comment:
in general we wouldn't expect a save_outputs function to return anything so it might make sense to rename or do something like 

output_paths = save_outputs(config, evaluation_results)
return output_paths